packages/did-mailto/test/did-mailto.spec.js

```diff
@@ -0,0 +1,88 @@
+import * as assert from 'assert'
+import * as didMailto from '../src/index.js'
+
+describe('did-mailto', () => {
+  testDidMailto(didMailto, async (name, test) => it(name, test))
+})
+
+/**
+ * @param {typeof didMailto} didMailto
+ * @param {import("./test-types").TestAdder} test
+ */
+function testDidMailto(didMailto, test) {
+  test('module is an object', async () => {
+    assert.equal(typeof didMailto, 'object')
+  })
+  for (const { email, did } of examples()) {
+    test(`fromEmail("${email}")`, async () => {
+      assert.deepStrictEqual(
+        didMailto.fromEmail(email),
+        did
+      )
+    })
+    test(`toEmail("${did}")`, async () => {
+      assert.deepStrictEqual(
+        didMailto.toEmail(did),
+        email
+      )
+    })
+    test(`toEmail(fromEmail("${email}"))`, async () => {
+      assert.deepStrictEqual(
+        didMailto.toEmail(didMailto.fromEmail(email)),
+        email
+      )
+    })
+    test(`fromEmail(toEmail("${did}"))`, async () => {
+      assert.deepStrictEqual(
+        didMailto.fromEmail(didMailto.toEmail(did)),
+        did
+      )
+    })
+  }
+  for (const email of validEmailAddresses()) {
+    test(`email("${email}")`, async () => {
+      assert.doesNotThrow(
+        () => didMailto.email(email),
+        'can parse to email'
+      )
+    })
+  }
+}
+
+function* examples() {
+  yield {
+    email: didMailto.email('example+123@example.com'),
+    did: didMailto.fromString('did:mailto:example.com:example%2B123'),
+  }
+  yield {
+    email: didMailto.email('"email@1"@example.com'),
+    did: didMailto.fromString(`did:mailto:example.com:%22email%401%22`),
+  }
+}
+
+function* validEmailAddresses() {
+  // https://gist.github.com/cjaoude/fd9910626629b53c4d25#file-gistfile1-txt-L5
+  yield* [
+    'email@example.com',
+    'firstname.lastname@example.com',
+    'email@subdomain.example.com',
+    'firstname+lastname@example.com',
+    'email@123.123.123.123',
+    'email@[123.123.123.123]',
+    '"email"@example.com',
+    '"email@1"@example.com',
+    '1234567890@example.com',
+    'email@example-one.com',
+    '_______@example.com',
+    'email@example.name',
+    'email@example.museum',
+    'email@example.co.jp',
+    'firstname-lastname@example.com',
```

Comment:
Can we test that following fail `email@` and `email@.com`

Reply:
Both of those strings would match the `EmailAddress` type I have in here.
I'm willing to make it more picky to disallow those two cases, but let's do it in another PR. I looked into doing more strict email address parsing, and to do it fully feels like more work than is really useful. (and if you need that, [they exist](https://www.npmjs.com/package/@sparser/email-address-parser)).